pkg/controller/backupstoragelocation_controller.go

```diff
@@ -0,0 +1,147 @@
+
+			if location.Name == r.StorageLocation.DefaultStorageLocation {
+				log.Warnf("The specified default backup location named %q is unavailable; for convenience, be sure to configure it properly or make another backup location that is available the default", r.StorageLocation.DefaultStorageLocation)
+			}
+
+			if err2 := r.StorageLocation.PatchStatus(location, velerov1api.BackupStorageLocationPhaseUnavailable); err2 != nil {
+				log.WithError(err).Errorf("Error updating backup location phase to %s", velerov1api.BackupStorageLocationPhaseUnavailable)
+				continue
+			}
+		} else {
+			log.Debug("Backup location verified and it is valid")
+			if err := r.StorageLocation.PatchStatus(location, velerov1api.BackupStorageLocationPhaseAvailable); err != nil {
+				log.WithError(err).Errorf("Error updating backup location phase to %s", velerov1api.BackupStorageLocationPhaseAvailable)
+				continue
+			}
+		}
+	}
+
+	if !anyVerified {
+		log.Info("No backup locations were ready to be verified")
+	}
+
+	r.logReconciledPhase(defaultFound, locationList, unavailableErrors)
+
+	return ctrl.Result{Requeue: true}, nil
+}
+
+func (r *BackupStorageLocationReconciler) logReconciledPhase(defaultFound bool, locationList velerov1api.BackupStorageLocationList, errs []string) {
+	var availableBSLs []*velerov1api.BackupStorageLocation
+	var unAvailableBSLs []*velerov1api.BackupStorageLocation
+	var unknownBSLs []*velerov1api.BackupStorageLocation
+	log := r.Log.WithField("controller", "backupstoragelocation")
+
+	for i, location := range locationList.Items {
+		phase := location.Status.Phase
+		switch phase {
+		case velerov1api.BackupStorageLocationPhaseAvailable:
+			availableBSLs = append(availableBSLs, &locationList.Items[i])
+		case velerov1api.BackupStorageLocationPhaseUnavailable:
+			unAvailableBSLs = append(unAvailableBSLs, &locationList.Items[i])
+		default:
+			unknownBSLs = append(unknownBSLs, &locationList.Items[i])
+		}
+	}
+
+	numAvailable := len(availableBSLs)
+	numUnavailable := len(unAvailableBSLs)
+	numUnknown := len(unknownBSLs)
+
+	if numUnavailable+numUnknown == len(locationList.Items) { // no available BSL
+		if len(errs) > 0 {
+			log.Errorf("Current backup storage locations available/unavailable/unknown: %v/%v/%v, %s)", numAvailable, numUnavailable, numUnknown, strings.Join(errs, "; "))
+		} else {
+			log.Errorf("Current backup storage locations available/unavailable/unknown: %v/%v/%v)", numAvailable, numUnavailable, numUnknown)
+		}
+	} else if numUnavailable > 0 { // some but not all BSL unavailable
+		log.Warnf("Invalid backup locations detected: available/unavailable/unknown: %v/%v/%v, %s)", numAvailable, numUnavailable, numUnknown, strings.Join(errs, "; "))
+	}
+
+	if !defaultFound {
+		log.Warnf("The specified default backup location named %q was not found; for convenience, be sure to create one or make another backup location that is available the default", r.StorageLocation.DefaultStorageLocation)
```

Comment:
This log message doesn't seem to read correctly.
Are we asking users to create a backup location named `default` and ensure that it's available?

Reply:
This is a warning that there's no bsl enabled as default. The warning suggests to create one, or change another bsl to be the default.

Reply:
Velero can operate without a default BSL specified if individual backups have one defined on them; it's mostly a convenience feature, as the message says.

[`StorageLocation`](https://github.com/vmware-tanzu/velero/blob/master/pkg/apis/velero/v1/backup.go#L80) is the field name on the `Backup` type, and `--storage-location` is the option on `velero backup create`. If the field is empty, then the default location is used, provided there is one.

Reply:
That sentence wasn't parsing correctly in my head. After the clarification it makes sense.